### ISSUE TITLE
Filter Menu - Improve form validation, add error messaging

### ISSUE THREAD

**Issue by adamkendis** [hackforla/311-data#581]
# Description
The Submit button is currently disabled until the user has selected all required fields. Other than the button being disabled, the user is provided no indication of required fields or incomplete selections. 

**Required Fields**
Map/Data Visualization: Date range, at least one NC, at least one request type
Comparison: Date range, both district sets, chart selection, at least one request type

## Action Items

- [x] Filter menu - add asterisks to filter subheaders to indicate required fields.
- [ ] Display error message(s) under subheaders for incomplete fields if user clicks Submit without making all required selections. Error messages should be red text.
- [x] Ensure Submit button can't fire a request off to the back end unless all required fields are completed.

## Resources
https://xd.adobe.com/view/c04c5a3e-519a-466a-6bbb-bc38c512e66e-b67d/
